### ISSUE TITLE
inline script is empty when minified

### ISSUE THREAD

**Issue by elgs** [parcel-bundler/parcel#1173]
| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           |1.7.0
| Node             |v9.11.1
| npm/Yarn         |npm 5.8.0
| Operating System |Linux archlinux 4.15.15-1-ARCH #1 SMP PREEMPT Sat Mar 31 23:59:25 UTC 2018 x86_64 GNU/Linux

The inline script inside index.html is empty when building with `parcel build index.html`, however, the inline script are correctly included when building with `parcel build --no-minify index.html`.

I have included a simplest example:

index.html
```html
<html lang="en">

<head>
    <script src="./app.js"></script>
</head>

<body>
    <script>
        console.log('hello');
        const p = new Point(10, 10);
    </script>
</body>

</html>
```

app.js
```javascript
class Point {
    constructor(x = 1, y = 1) {
        this.x = x;
        this.y = y;
    }

    move(x, y) {
        this.x = x;
        this.y = y;
    }
}

window.Point = Point;
```

**Comment by DeMoorJasper** [comment]
That's weird normally source should be added correctly

**Comment by elgs** [comment]
What's interesting is if I comment out the line `const p = new Point(10, 10);`, it works correctly with or without `--no-minify`.

**Comment by DeMoorJasper** [comment]
Intresting, thanks for the debugging. Now we have a good minimal reproduction script.
Feel free to dig deeper into this

**Comment by elgs** [comment]
Thanks. Any chance to take a look at this question. This virtually makes this version of parcel not usable.

**Comment by DeMoorJasper** [comment]
I think I've found a fix for this, it's a htmlnano issue 100% sure as we don't touch inline js.
You could turn of minification for inline js using the following htmlnano config (`.htmlnanorc.js`):
```Javascript
module.exports = {
  collapseWhitespace: 'conservative',
  minifyCss: {
    safe: true
  },
  minifyJs: false // This will disable js minification
}
```

**Comment by elgs** [comment]
It seems the problem still exists in 1.7.1.

**Comment by DeMoorJasper** [comment]
@elgs The fixes in 1.7.1 didn't change the htmlnano config, could you try with the config I proposed?

**Comment by elgs** [comment]
Yes I can. But they are some workarounds. I cannot put the inline code in the bundle because the inline code is sample code which doesn’t belong to the bundle. As to editing the config file, the minification should be disabled by default if it doesn’t work.

**Comment by elgs** [comment]
@DeMoorJasper thanks I will try. But where is `.htmlnanorc.js`?

**Comment by DeMoorJasper** [comment]
@elgs u create it in the root of your project

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by elgs** [comment]
@DeMoorJasper thanks it works for me. Would you please talk a little more about:

```javascript
minifyCss: {
    safe: true
},
```
It seems I didn't see the `safe` option anywhere.

**Comment by DeMoorJasper** [comment]
@elgs It's a cssnano options that prevents unsafe minification which could lead to unexpected css output.

**Comment by elgs** [comment]
It seems I didn't see anywhere talking about the `safe` option in cssnano's website. Would you mind to share a link about it?

Also I'm wondering whether it is appropriate to close this bug at this moment of time. It's like a car manufacture closes a ticket because the problem is from the part provider before the problem is solved, unless this problem is not considered a problem.

**Comment by DeMoorJasper** [comment]
@elgs We provided a fix while htmlnano tries to resolve it (unless u didn't file a bug report there), there is not much more we can do. Doesn't seem like it happens that often so including it as standard config, doesn't seem like the obvious choice.

Not sure where cssnano's safe option is defined in their docs, but it describes what it does in the readme. They made it the default behaviour recently as I can see, so it might even be possible that they removed it from the docs, as it's the default now.

**Comment by elgs** [comment]
@DeMoorJasper thanks for the explanation. I haven't file a but report to `htmlnano`. I am too lazy to prepare a use case for their project from scratch. However, I will share this thread with them. I guess this thread should provide enough information for them to address this problem, if they want, of course.

**Comment by maltsev** [comment]
Thanks for submitting the bug! I published new version 0.1.8 with the fix.

**Comment by elgs** [comment]
Thanks @maltsev!

@DeMoorJasper I saw the dependencies in the package.json in the`parcel-bundle` project like this:

`"htmlnano": "^0.1.7",`

I did `sudo npm update -g`, but it seems it didn't upgrade `htmlnano` to 0.1.8.
